rust/lancedb/resources/remote-catalog-open-api.yaml

```diff
@@ -0,0 +1,284 @@
+openapi: 3.1.1
+info:
+  title: LanceDB REST Catalog API
+  version: 0.0.1
+  description: REST interface for managing LanceDB databases and tables
+
+paths:
+  /v1/databases:
+    get:
+      tags: [Database]
+      summary: List databases
+      parameters:
+        - $ref: '#/components/parameters/startAfter'
+        - $ref: '#/components/parameters/limit'
+      responses:
+        200:
+          description: Database list response
+          content:
+            application/json:
+              schema:
+                $ref: '#/components/schemas/DatabaseList'
+
+    post:
+      tags: [Database]
+      summary: Create new database
+      requestBody:
+        required: true
+        content:
+          application/json:
+            schema:
+              $ref: '#/components/schemas/CreateDatabaseRequest'
+      responses:
+        201:
+          description: Database created
+          headers:
+            Location:
+              description: URI of created database
+              schema: {type: string}
+
+    delete:
+      tags: [ Database ]
+      summary: Delete all databases
```

Comment:
One thing I like about Gravitino's naming convention is that they use SQL verbs instead of REST verbs to describe the operation, for example use `AlterDatabase` instead of `UpdateDatabase`, `DropDatabase` instead of `DeleteDatabase`, I think that helps people with data background to understand the behavior better. What do you think?